apps/src/code-studio/assets/SpriteUpload.jsx

```diff
@@ -44,26 +57,48 @@ export default class SpriteUpload extends React.Component {
         break;
     }
 
+    // The sprite and metadata JSON should have the same name, but different file extensions
+    let jsonDestination = destination.replace('.png', '.json');
+
     return uploadSpriteToAnimationLibrary(
       destination,
       fileData,
       this.onSuccess,
       this.onError
+    ).then(() =>
+      uploadMetadataToAnimationLibrary(
```

Comment:
what happens if `uploadSpriteToAnimationLibrary` fails? is this method still invoked?

Reply:
we could also create an `onSpriteUploadSuccess` method that calls `this.onSuccess` and `uploadMetadataToAnimationLibrary` so that we only upload the metadata if the sprite upload succeeds

Reply:
This is a bit complicated since fetch doesn't reject for HTTP error statuses, so we'll need to handle the error case in onSuccess and onError. Since fetch returns a promise, I have a preference for finding a way to use resolve/reject to handle this, but promises are hard for my brain, so I need to play around with this for a bit more.

Reply:
ahh, i see. i forgot about that -- we should refactor `animationLibraryApi` functions to only call `onSuccess` if the request was successful. then you won't have to handle the error case in both places and it won't be confusing for new users of those functions

Reply:
Based on this documentation: "An accurate check for a successful fetch() would include checking that the promise resolved, then checking that the Response.ok property has a value of true." (https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch), there still are three states - success, !response.ok, and promise rejected. I don't believe there is a way to handle both the error in catch and !response.ok, with what we currently have. I'll continue to work on finding a way to get these all aligned, but, especially since this is an internal tool with (in the long-run) pretty limited use, we might hit the point where it just makes sense to handle all three states individually.